### PR TITLE
docs(highlight): fix type annotations

### PR DESCRIPTION
**Pull request by tomtomjhj**

### CODE REVIEW

File: runtime/lua/vim/highlight.lua
Comment (lewis6991) [inline runtime/lua/vim/highlight.lua:43]: We should change all `number`s to `integer`.

Comment (tomtomjhj) [reply runtime/lua/vim/highlight.lua]: Fixed, and changed `number` to `integer` in other lua files too.

**Comment by lewis6991** [comment]
Don't forget to run `./scripts/gen_vimdoc.py`

**Comment by tomtomjhj** [comment]
It seems that gen_vimdoc.py doesn't generate docs for highlight.lua. I'll take a look later.

**Comment by tomtomjhj** [comment]
Tweaked gen_vimdoc.py to generate the docs for vim.hightlight. Please have a look.

**Comment by clason** [comment]
Could you reorder the sections so that they're alphabetical again? And the docs for  `on_yank()` are a bit too terse now; it doesn't have to be quite as verbose as my old text, but it should be complete sentences...

(While you're at it, you could convert the examples to Lua using `vim.api.nvim_create_autocmd`.)

**Comment by tomtomjhj** [comment]
> And the docs for `on_yank()` are a bit too terse now; it doesn't have to be quite as verbose as my old text, but it should be complete sentences...
> 
> (While you're at it, you could convert the examples to Lua using `vim.api.nvim_create_autocmd`.)

done



> Could you reorder the sections so that they're alphabetical again?

If you mean `section_order` in gen_vimdoc.py, the sections were not in alphabetical order in the first place

**Comment by tomtomjhj** [comment]
can we merge this

**Comment by clason** [comment]
I'll defer to @lewis6991 here, but this is touching a few areas that are under heavy development right now, and I'd prefer _those_ PRs not to be saddled with a merge conflict (I'm looking at you, LSP semantic tokens...)

So if you're OK with this, I'd prefer to park this until the big PRs are in to make sure we don't regress anything, and then merge this (after a rebase/final sweep) for the 0.9 feature freeze?

**Comment by lewis6991** [comment]
Yes, that makes sense to me. In the future it would be best to limit the size of PR's.

**Comment by clason** [comment]
It would at least help to split the PR into (self-contained) commits focusing on a single "subsystem" like LSP, treesitter, core... Then we could split out and merge those that do not conflict with ongoing work.

(As @lewis6991 said, smaller is better for PRs; it's enough to mention in the PR description that this is part of a larger effort.)

**Comment by tomtomjhj** [comment]
> I'll defer to @lewis6991 here, but this is touching a few areas that are under heavy development right now, and I'd prefer _those_ PRs not to be saddled with a merge conflict (I'm looking at you, LSP semantic tokens...)

Makes sense. Then what about just merging the commits excluding the last one (number -> integer)?

**Comment by clason** [comment]
If you squash them down and drop the last one from this PR (and open a new one with it, either now or later), sure.

**Comment by tomtomjhj** [comment]
I'd like to merge these two commits without squashing in this PR:
* docs(highlight): use gen_vimdoc for vim.highlight
* docs(highlight): fix type annotations

I'll make a separate PR for this commit
* docs(lsp): type annotation for lsp.client

Finally, I'll break this commit into smaller commit involving a single component, and make PR separetely.
* docs: number → integer

File: runtime/lua/vim/highlight.lua
Comment (clason) [inline runtime/lua/vim/highlight.lua:92]:
```suggestion
vim.api.nvim_create_autocmd("TextYankPost", {
```

File: runtime/lua/vim/highlight.lua
Comment (clason) [inline runtime/lua/vim/highlight.lua:100]:
```suggestion
vim.api.nvim_create_autocmd("TextYankPost", {
```

**Review by clason — CHANGES_REQUESTED** [review]
Not crazy about the fake `priorities()` function, which is kinda misleading.

Also, might make sense to wait for https://github.com/neovim/neovim/pull/22498?

(But in general +1 for using gen_vimdoc instead of hand-writing docs.)

File: runtime/lua/vim/highlight.lua
Comment (tomtomjhj) [reply runtime/lua/vim/highlight.lua:92]: Do you mean this?
```suggestion
--- vim.api.nvim_create_autocmd("TextYankPost", {
```

**Comment by tomtomjhj** [comment]
> Also, might make sense to wait for https://github.com/neovim/neovim/pull/22498?

oh that looks good. I'll defer "docs(highlight): use gen_vimdoc for vim.highlight" to another PR and leave only "docs(highlight): fix type annotations" for this PR.

File: runtime/lua/vim/highlight.lua
Comment (clason) [reply runtime/lua/vim/highlight.lua:92]: Sorry, the Github diffview tricked me into seeing an empty line where there was none. Disregard, please.

**Review by clason — APPROVED** [review]

**Pull request merged**
